tests/backup/backup_resiliency_test.go

```diff
@@ -1558,3 +1558,284 @@ var _ = Describe("{ScaleDownPxBackupPodWhileBackupAndRestoreIsInProgress}", func
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Taking backup %s of application- %s", backupName, namespace))
+				backupNames = append(backupNames, backupName)
+
+				log.Infof("Taking schedule backup of application with rule's")
+				scheduleName = fmt.Sprintf("%s-%s-%v", BackupNamePrefix, namespace, time.Now().Unix())
+				schPolicyUid, _ = Inst().Backup.GetSchedulePolicyUid(orgID, ctx, periodicPolicyName)
+				preRuleUid, _ = Inst().Backup.GetRuleUid(orgID, ctx, preRuleNameListMap[namespace])
+				postRuleUid, _ = Inst().Backup.GetRuleUid(orgID, ctx, postRuleNameListMap[namespace])
+				err = CreateScheduleBackup(scheduleName, SourceClusterName, bkpLocationName, backupLocationUID, []string{namespace},
+					labelSelectors, orgID, preRuleNameListMap[namespace], preRuleUid, postRuleNameListMap[namespace], postRuleUid, periodicPolicyName, schPolicyUid, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verification of creating schedule backup with schedule name - %s", scheduleName))
+				schBackupName, err = GetFirstScheduleBackupName(ctx, scheduleName, orgID)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching the name of the first schedule backup - %s", schBackupName))
+				backupNames = append(backupNames, schBackupName)
+				scheduleNames = append(scheduleNames, scheduleName)
+				backupNamesMap[namespace] = backupNames
+				threadCount = threadCount + len(backupNames)
+			}
+			log.Infof("The Map of backups taken are: %v", backupNamesMap)
+		})
+
+		var sem = make(chan struct{}, threadCount)
+		var wg sync.WaitGroup
+		Step("Restoring the backed up applications", func() {
+			log.InfoD("Restoring the backed up applications")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range appNamespaces {
+				namespaceMapping := make(map[string]string)
+				for _, backupName := range backupNamesMap[namespace] {
+					sem <- struct{}{}
+					restoreName = fmt.Sprintf("%s-%s", restoreNamePrefix, backupName)
+					restoreNames = append(restoreNames, restoreName)
+					time.Sleep(1 * time.Second)
+					namespaceMapping[namespace] = fmt.Sprintf("new-namespace-%v", time.Now().Unix())
+					wg.Add(1)
+					go func(restoreName string, backupName string, namespaceMapping map[string]string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						defer func() { <-sem }()
+						_, err = CreateRestoreWithoutCheck(restoreName, backupName, namespaceMapping, destinationClusterName, orgID, ctx)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Restoring the backup %s with name %s", backupName, restoreName))
+					}(restoreName, backupName, namespaceMapping)
+				}
+			}
+			wg.Wait()
+			log.Infof("The list of restores taken are: %v", restoreNames)
+			log.InfoD("Sleeping for 20 seconds so that the request reaches stork and the restore process is started")
+			time.Sleep(20 * time.Second)
```

Comment:
Wouldn't it be better to check every second for status of a restore to be `InProgress`, and as soon as that happen, one can initiate a delete?  (I have seen a restore finish in 20 seconds, whereas I think this testcase requires it to be `InProgress` i.e. it should be running when deleted, right?)

Or is that unnecessary complexity and sleeping is a simpler alternative?

Reply:
Added logic to tiggger delete while restore status is in progress.